### ISSUE TITLE
Breaking change in addon API inherited from (`cryptography`?)

### ISSUE THREAD

**Issue by Prinzhorn** [mitmproxy/mitmproxy#6750]
#### Problem Description

This is an interesting problem. The CHANGELOG for mitmproxy 10 claims that the only breaking change for addons are `connection.Client/Server` constructor. However, `Cert.altnames` has also changed. I assume this was inherited from an upgrade to `cryptography`? 

It would be easy for me to update the CHANGELOG, but this is a general problem how we can avoid breaking addons in the future when inheriting changes from other libraries.

#### Steps to reproduce the behavior:

addon.py

```py
def tls_established_server(data):
    server = data.conn

    for cert in server.certificate_list:
        for name in cert.altnames:
            print(name)
```

1.  `mitmdump --scripts addon.py`
2.  `curl --insecure --proxy localhost:8080 https://example.com/`

# Mitmproxy 9 output

```
www.example.org
example.net
example.edu
example.com
example.org
www.example.com
www.example.edu
www.example.net
```

# Mitmproxy 10 output

```
<DNSName(value='www.example.org')>
<DNSName(value='example.net')>
<DNSName(value='example.edu')>
<DNSName(value='example.com')>
<DNSName(value='example.org')>
<DNSName(value='www.example.com')>
<DNSName(value='www.example.edu')>
<DNSName(value='www.example.net')>
```

**Comment by mhils** [comment]
The change was not directly inherited from cryptography but implemented in https://github.com/mitmproxy/mitmproxy/commit/b659ea0101ed6af8dc5dc227d8e71aafcda12d75. Based on what you are saying we should have a better CHANGELOG entry? Feel free to propose something, otherwise I'll try to come up with a better version.

**Comment by mhils** [comment]
Ah, now I think I see what you mean. This should be on https://docs.mitmproxy.org/stable/addons-api-changelog/, right?

**Comment by Prinzhorn** [comment]
> The change was not directly inherited from cryptography but implemented in [b659ea0](https://github.com/mitmproxy/mitmproxy/commit/b659ea0101ed6af8dc5dc227d8e71aafcda12d75). Based on what you are saying we should have a better CHANGELOG entry? Feel free to propose something, otherwise I'll try to come up with a better version.

I see. What I meant is that I would expect this to be listed under https://github.com/mitmproxy/mitmproxy/blob/main/CHANGELOG.md#breaking-changes . When I upgrade major versions of anything I mostly look into that section and don't screen the rest for additional breaking changes.

I think 

> This should be on https://docs.mitmproxy.org/stable/addons-api-changelog/, right?

Didn't know this existed :smile:
